src/hooks/useUserBalanceForPaymentTokenByDomain.ts

```diff
@@ -0,0 +1,64 @@
+//- React Imports
+import { useQuery } from 'react-query';
+
+//- Hooks Imports
+import { useZnsSdk } from './useZnsSdk';
+
+//- Constants Imports
+import { PAYMENT_TOKENS, DEFAULT_NETWORK } from '../constants/network';
+
+//- Hooks
+import { useBuyDomain } from './useBuyDomain';
+
+//- Utils
+import { formatEther } from '@ethersproject/units';
+
+//- Types
+export type UserBalanceForPaymentTokenByDomainReturn = {
+	isLoading: boolean;
+	balance?: string;
+};
+
+export const useUserBalanceForPaymentTokenByDomain = (
+	account: string,
+	chainId: number,
+): UserBalanceForPaymentTokenByDomainReturn => {
+	// SDK
+	const sdk = useZnsSdk();
+
+	// Domain Name
+	const { domainName } = useBuyDomain();
+
+	// Payment Token
+	const paymentToken = PAYMENT_TOKENS[chainId];
```

Comment:
nit: it could make sense to move this into the `useBuyDomain` hook - no biggie though.